dev/PushNotifications/PushNotificationBackgroundTask.cpp

```diff
@@ -0,0 +1,25 @@
+﻿#pragma once
+
+#include "pch.h"
+
+#include "PushNotificationBackgroundTask.h"
+
+#include <winrt/base.h>
+#include "PushNotificationReceivedEventArgs.h"
+#include "externs.h"
+
+namespace winrt
+{
+    using namespace Windows::ApplicationModel::Background;
+    using namespace Microsoft::Windows::PushNotifications;
+}
+
+wil::unique_handle g_waitHandleForArgs;
+winrt::PushNotificationReceivedEventArgs g_activatedEventArgs{ nullptr };
```

Comment:
@oldnewthing How can a cleanup object be registered? Do you have any pointers to code that can address this concern?

Reply:
The other thing to mention here. If a game were to disable COM while it is running, it would break the Push scenario which kinda seems counter-intuitive to having the feature in the first place. So I believe that the Reunion app will have to keep COM server running always till rundown occurs.

Reply:
As I noted in the original comment, you can use the COM static store to learn when COM is uninitialized.

What if an app unregisters its push listener before uninitializing COM? Maybe it doesn't want to listen to push notifications while the game is in progress (uses it only for matchmaking say).

Reply:
Thats a valid scenario for a game developer. 
RE. COM static store, I guess my question was a pointer to the APIs in MSDN. We can take it from there.

Reply:
"COM static store" is just a nickname. Its official name is the "CoreApplication.Properties PropertySet".

Reply:
I guess you are inferring to this?
https://docs.microsoft.com/en-us/uwp/cpp-ref-for-winrt/static-lifetime

Reply:
That is an example of another component that uses the COM static store. Here's [some more code that uses the COM static store](https://devblogs.microsoft.com/oldnewthing/20210208-00/?p=104812).

Reply:
Thanks @oldnewthing 
@pmpurifoy Lets get the PushNotificationReceivedEventArgs to be part of the COM static store.

For the statics below(s_typedEventHandler), we can solve the lifetime problem by refactoring the test code to ensure that the channel object isn't scoped to the lifetime of the lambda. So you can essentially eliminate that static and the lock associated with it.